src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticLineEnderTests.cs

```diff
@@ -841,6 +846,668 @@ void Method()
+        {
+            Test($@"
+public {typeKeyword} Bar2
+{{
+    $$
+}}
+public {typeKeyword} Bar
+{{
+}}", $@"
+public {typeKeyword} B$$ar2
+public {typeKeyword} Bar
+{{
+}}");
+        }
+
+        [WpfTheory]
+        [InlineData("namespace")]
+        [InlineData("class")]
+        public void TestNestedNamespaceAndTypeDeclaration(string typeKeyword)
+        {
+            Test($@"
+public {typeKeyword} Bar1
+{{
+    public {typeKeyword} Bar2
+    {{
+        $$
+    }}
+}}",
+$@"
+public {typeKeyword} Bar1
+{{
+    public {typeKeyword} B$$ar2
+}}");
+        }
+
+        [WpfTheory]
+        [InlineData("namespace")]
+        [InlineData("class")]
+        [InlineData("struct")]
+        [InlineData("record")]
+        [InlineData("enum")]
+        [InlineData("interface")]
+        public void TestBaseTypeDeclarationAndNamespaceWithOpenBrace(string typeKeyword)
+        {
+            Test($@"
+public {typeKeyword} Bar {{
+    $$", $@"
+public {typeKeyword} B$$ar {{");
+        }
+
+        [WpfTheory]
+        [InlineData("namespace")]
+        [InlineData("class")]
+        [InlineData("struct")]
+        [InlineData("record")]
+        [InlineData("enum")]
+        [InlineData("interface")]
+        public void TestValidTypeDeclarationAndNamespace(string typeKeyword)
+        {
+            Test($@"public {typeKeyword} Bar {{}}
+$$",
+                $@"public {typeKeyword} Ba$$r {{}}");
+        }
+
+        [WpfFact]
+        public void TestMethod()
+        {
+            Test(@"
+public class Bar
+{
+    void Main()
+    {
+        $$
+    }
+}", @"
+public class Bar
+{
+    void Ma$$in()
+}");
+        }
+
+        [WpfFact]
+        public void TestValidMethodInInterface()
+        {
+            Test(@"
+public interface Bar
+{
+    void Main();
+    $$
+}", @"
+public interface Bar
+{
+    void Mai$$n();
+}");
+        }
+
+        [WpfFact]
+        public void TestValidLocalFunction()
+        {
+            Test(@"
+public class Bar
+{
+    void Main()
+    {
+        void Local()
+            $$
+        {
+        }
+    }
+}", @"
+public class Bar
+{
+    void Main()
+    {
+        void Loc$$al()
+        {
+        }
+    }
+}");
+        }
+
+        [WpfFact]
+        public void TestLocalFunction()
+        {
+            Test(@"
+public class Bar
+{
+    void Main()
+    {
+        void Local()
+        {
+            $$
+        }
+    }
+}", @"
+public class Bar
+{
+    void Main()
+    {
+        void Loca$$l()
+    }
+}");
+        }
+
+        [WpfFact]
+        public void TestIndexer1()
+        {
+            Test(@"
+public class Bar
+{
+    public int this[int i]
+    {
+        $$
+    }
+}", @"
+public class Bar
+{
+    public int thi$$s[int i]
+}");
+        }
+
+        [WpfFact]
+        public void TestIndexer2()
+        {
+            Test(@"
+public class Bar
+{
+    public int this[int i]
+    {
+        $$
+    }
+    void Main() {}
+}", @"
+public class Bar
+{
+    public int thi$$s[int i]
```

Comment:
💭 The markup parser has APIs that are able to understand multiple positions (i.e. multiple instances of `$$` in the text). It would be nice to modify `Test` to use this API when processing the input text, and then automatically verify _each_ of the positions produces the same output. Then we could do this:

```suggestion
    pu$$blic int thi$$s[int $$i]$$
```

Reply:
Support for this is now included in the PR, so the new tests can be updated to use more positions. 😄